apps/src/Register3D.cpp

```diff
@@ -0,0 +1,185 @@
+        return EXIT_FAILURE;
+    }
+
+    auto cvMovingImage = cv::imread(movingImageFileName);
+    // Read the two images
+    movingImage =
+        itk::OpenCVImageBridge::CVMatToITKImage<rt::Image8UC3>(cvMovingImage);
+
+    // Ignore spacing information
+    fixedImage->SetSpacing(1.0);
+    movingImage->SetSpacing(1.0);
+
+    // Read the landmarks file
+    rt::LandmarkIO landmarkReader(landmarksFileName);
+    landmarkReader.setFixedImage(fixedImage);
+    landmarkReader.setMovingImage(movingImage);
+    landmarkReader.read();
+    auto fixedLandmarks = landmarkReader.getFixedLandmarks();
+    auto movingLandmarks = landmarkReader.getMovingLandmarks();
+
+    ///// Landmark Registration /////
+    printf("Running landmark registration...\n");
+
+    // Generate the landmark transform
+    rt::LandmarkRegistration landmark;
+    landmark.setFixedLandmarks(fixedLandmarks);
+    landmark.setMovingLandmarks(movingLandmarks);
+    auto ldmTransform = landmark.compute();
+
+    // Apply it to the image
+    auto tmpMovingImage =
+        rt::ImageTransformResampler(fixedImage, movingImage, ldmTransform);
+
+    ///// Deformable Registration /////
+    printf("Running deformable registration...\n");
+    rt::DeformableRegistration deformable;
+    deformable.setFixedImage(fixedImage);
+    deformable.setMovingImage(tmpMovingImage);
+    auto deformTransform = deformable.compute();
+
+    ////// USED FOR READING IN TRANSFORM FILE //////
+    /*TransformReader::Pointer transformReader = TransformReader::New();
+    transformReader->SetFileName(transformIn);
+    try {
+        // Software Guide : BeginCodeSnippet
+        transformReader->Update();
+    }
+    catch(itk::ExceptionObject & excp) {
+        std::cerr << "Error while reading the transform file" << std::endl;
+        std::cerr << excp << std::endl;
+        std::cerr << "[FAILED]" << std::endl;
+        return EXIT_FAILURE;
+    }
+    const TransformReader::TransformListType *transforms =
+    transformReader->GetTransformList();
+
+    CompositeTransform::Pointer compositeTrans;
+    TransformReader::TransformListType::const_iterator it = transforms->begin();
+    if(!strcmp((*it)->GetNameOfClass(),"CompositeTransform")) {
+        compositeTrans = static_cast< CompositeTransform* >((*it).GetPointer());
+        compositeTrans->Print(std::cout);
+    }*/
+
+    ///// Combine transforms /////
+    CompositeTransform::Pointer compositeTrans = CompositeTransform::New();
+    compositeTrans->AddTransform(ldmTransform);
+    compositeTrans->AddTransform(deformTransform);
+
+    printf("Finished registration\n\n");
+    auto origUVMap = reader.getUVMap();
+    auto newUVMap = volcart::UVMap();
+
+    for (auto point = origMesh->GetPoints()->Begin();
+         point != origMesh->GetPoints()->End(); ++point) {
+        // assign the point
+        auto origUV = origUVMap.get(point->Index());
+        if (origUV != VC_UVMAP_NULL_MAPPING) {
+            itk::Point<double, 2> in;
```

Comment:
You have the cv::Mat's for these images, so use cv functions instead. It makes the math easier:
```
auto in = origUV * {cvFixed.cols, cvFixed.rows};
auto out = compositeTrans->TransformPoint(in.val);
cv::Vec2d newUV{ out[0] / (cvMoving.cols - 1), out[1] / (cvMoving.rows - 1) };
```
Note the `- 1` for each dimension. You want to divide by your largest possible index. Since images are zero indexed, that's one less than the size in each dimension.